File: lang_java/Petshop.java
public class Petshop{
// membuat class Petshop

    private String idProduk; // atribut id produk
    private String namaProduk; // atribut nama produk
    private int hargaProduk; // atribut harga produk
    private int stokProduk; // atribut stok produk

    public Petshop(){
        // konstruktor kosong
    }

    public Petshop(String idProduk, String namaProduk, int hargaProduk, int stokProduk){
        // konstruktor langsung mengisi atribut
        this.idProduk = idProduk;
        this.namaProduk = namaProduk;
        this.hargaProduk = hargaProduk;
        this.stokProduk = stokProduk;
    }

    public void setIdProduk(String idProduk){
        // setter untuk mengisi id produk
        this.idProduk = idProduk;
    }

    public String getIdProduk(){
        // getter untuk mendapatkan id produk
        return idProduk;
    }

    public void setNamaProduk(String namaProduk){
        // setter untuk mengisi nama produk
        this.namaProduk = namaProduk;
    }

    public String getNamaProduk(){
        // getter untuk mendapatkan nama produk
        return namaProduk;
    }

    public void setHargaProduk(int hargaProduk){
        // setter untuk mengisi harga produk
        this.hargaProduk = hargaProduk;
    }

    public int getHargaProduk(){
        // getter untuk mendapatkan harga produk
        return hargaProduk;
    }

    public void setStokProduk(int stokProduk){
        // setter untuk mengisi stok produk
        this.stokProduk = stokProduk;
    }

    public int getStokProduk(){
        // getter untuk mendapatkan stok produk
        return stokProduk;
    }

}

File: lang_cpp/Petshop.cpp
#include <iostream> // import library input/output
#include <string> // import library string

using namespace std; // menggunakan standard namespace

class Petshop{
// membuat class petshop

    // deklarasi atribut private
    private:
        string idProduk;
        string namaProduk;
        int hargaProduk;
        int stokProduk;

    // deklarasi method public
    public:

        // konstruktor tanpa parameter
        Petshop(){
        };

        // konstruktor dengan parameter
        Petshop(string idProduk, string namaProduk, int hargaProduk, int stokProduk){
            this->idProduk = idProduk;
            this->namaProduk = namaProduk;
            this->hargaProduk = hargaProduk;
            this->stokProduk = stokProduk;
        }

        // setter untuk id petshop
        void setIdProduk(string idProduk){
            this->idProduk = idProduk;
        }

        // getter untuk id petshop
        string getIdProduk(){
            return idProduk;
        }

        // setter untuk nama produk
        void setNamaProduk(string namaProduk){
            this->namaProduk = namaProduk;
        }

        // getter untuk nama produk
        string getNamaProduk(){
            return namaProduk;
        }

        // setter untuk harga produk
        void setHargaProduk(int hargaProduk){
            this->hargaProduk = hargaProduk;
        }

        // getter untuk harga produk
        int getHargaProduk(){
            return hargaProduk;
        }

        // setter untuk stok produk
        void setStokProduk(int stokProduk){
            this->stokProduk = stokProduk;
        }

        // getter untuk stok produk
        int getStokProduk(){
            return stokProduk;
        }

        ~Petshop(){
            // destruktor
        }
  
};